internal/resource/config/certificate/certificate_ca_data_source.go

```diff
@@ -0,0 +1,222 @@
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"signature_algorithm": schema.StringAttribute{
+				Description: "The signature algorithm",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"version": schema.Int64Attribute{
+				Description: "The X.509 version to which the item conforms",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"sha1_fingerprint": schema.StringAttribute{
+				Description: "SHA-1 fingerprint in Hex encoding",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"sha256_fingerprint": schema.StringAttribute{
+				Description: "SHA-256 fingerprint in Hex encoding",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+			},
+			"status": schema.StringAttribute{
+				Description: "Status of the item.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"VALID", "EXPIRED", "NOT_YET_VALID", "REVOKED"}...),
+				},
+			},
+			"crypto_provider": schema.StringAttribute{
+				Description: "Cryptographic Provider. This is only applicable if Hybrid HSM mode is true.",
+				Required:    false,
+				Optional:    false,
+				Computed:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"LOCAL", "HSM"}...),
+				},
+			},
+		},
+	}
+	id.DataSourceSchema(&schemaDef, true, "The persistent, unique ID for the certificate")
+	resp.Schema = schemaDef
+}
+
+// Metadata returns the data source type name.
+func (r *certificatesDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_certificate_ca"
+}
+
+// Configure adds the provider configured client to the data source.
+func (r *certificatesDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+}
+
+// Read a CertificateResponse object into the model struct
+func readCertificateResponseDataSource(ctx context.Context, r *client.CertView, state *certificatesDataSourceModel, expectedValues *certificatesDataSourceModel, diagnostics *diag.Diagnostics) {
+	state.Id = internaltypes.StringTypeOrNil(r.Id, false)
+	state.SerialNumber = internaltypes.StringTypeOrNil(r.SerialNumber, false)
+	state.SubjectDN = internaltypes.StringTypeOrNil(r.SubjectDN, false)
+	state.SubjectAlternativeNames = internaltypes.GetStringSet(r.SubjectAlternativeNames)
+	state.IssuerDN = internaltypes.StringTypeOrNil(r.IssuerDN, false)
+	state.ValidFrom = types.StringValue(r.ValidFrom.Format(time.Now().String()))
+	state.Expires = types.StringValue(r.Expires.Format(time.Now().String()))
```

Comment:
How does this time.Now() stuff work? Is that going to change these values every time the data source runs a Read?